app/models/pq_scopes.rb

```diff
@@ -10,97 +10,148 @@ def not_seen_by_finance
     where(seen_by_finance: false)
   end
 
-  def accepted_in(aos)
-    joins(:action_officers_pqs).where(action_officers_pqs: {
-      response: 'accepted',
-      action_officer_id: aos 
-    })
-  end
-
-  def ministers_by_progress(ministers, progresses)
-    includes(:progress, :minister).
-      where(progress_id: progresses, minister_id: ministers).
-      group(:minister_id, :progress_id).
-      count
+  # Returns a hash with the PQ state as key and count by minister_id as value.
+  #
+  # It includes only PQs that are in the in progress states (PQState::IN_PROGRESS),
+  # and ministers that are active.
+  #
+  # @returns [Hash[String, Hash]]
+  #
+  # Example:
+  #
+  #     {
+  #       "no_response" => {
+  #         43 => 1,
+  #         44 => 10
+  #         48 => 2
+  #       },
+  #       "with_pod" => {
+  #         44 => 2
+  #       }
+  #     }
+  def in_progress_by_minister
+    select("minister_id, state, count(*)")
+      .joins(:minister)
+      .where(state: PQState::IN_PROGRESS)
+      .where('ministers.deleted = false')
+      .group(:state, :minister_id)
+      .reduce({}) { |acc, r|
+        h = { r.minister_id => r.count }
+        acc.merge(r.state => h ) { |_, old_v, new_v| old_v.merge(new_v) }
+      }
+  end
+
+  # Returns a hash with the PQ state as key and count by press_desk_id as value.
+  #
+  # It will filter out PQs that are below the PQState::DRAFT_PENDING state,
+  # and press desks that are not active.
+  #
+  def accepted_by_press_desk
```

Comment:
should we return a list of Pqs here and then reduce within the controller? Else rename function as implies PQ list returned